pkg/discovery/util/node_util.go

```diff
@@ -59,3 +59,13 @@ func GetNodeIP(node *api.Node) (string, error) {
 	}
 	return "", fmt.Errorf("Node %v has no valid hostname and/or IP address: %v", node.Name, ip)
 }
+
+// Check whether the node is a master
+func NodeIsMaster(node *api.Node) bool {
+	return detectors.IsMasterDetected(node.Name, node.ObjectMeta.Labels)
+}
+
+// Returns whether the node is controllable
+func NodeIsControllable(node *api.Node) bool {
+	return !NodeIsMaster(node)
```

Comment:
Add some logging here to help future troubleshooting when we detect the master.

Reply:
Although NodeIsMaster already generated logging output, I changed that log message to V4 and added new log message here.